File: BehavioralDesignPattern/ChainOfResponsibilityPattern/java/LoggerChain.java
package BehavioralDesignPattern.ChainOfResponsibilityPattern.java;

import BehavioralDesignPattern.ChainOfResponsibilityPattern.java.AbstractHandler.Logger;
import BehavioralDesignPattern.ChainOfResponsibilityPattern.java.ConcreteHandler.DebugLogger;
import BehavioralDesignPattern.ChainOfResponsibilityPattern.java.ConcreteHandler.ErrorLogger;
import BehavioralDesignPattern.ChainOfResponsibilityPattern.java.ConcreteHandler.InfoLogger;

public class LoggerChain {
    public static Logger getLoggerChain(){
        Logger InfoLogger = new InfoLogger();
        Logger DebugLogger = new DebugLogger();
        Logger ErrorLogger = new ErrorLogger();

        InfoLogger.setNextHandler(DebugLogger);
        DebugLogger.setNextHandler(ErrorLogger);
        ErrorLogger.setNextHandler(null);
        return InfoLogger;
    }
}

File: BehavioralDesignPattern/ChainOfResponsibilityPattern/python/LoggerChain.py
from ConcreteHandler.DebugLogger import DebugLogger
from ConcreteHandler.ErrorLogger import ErrorLogger
from ConcreteHandler.InfoLogger import InfoLogger


class LoggerChain:
    @staticmethod
    def getLoggerChain():
        infoLogger = InfoLogger()
        debugLogger = DebugLogger()
        errorLogger = ErrorLogger()

        infoLogger.setNextLogger(debugLogger)
        debugLogger.setNextLogger(errorLogger)
        return infoLogger